### PR TITLE
Task sf/j228 add follow user functionality

### PR DESCRIPTION
**Pull request by Sayonarko**
user-profile

### CODE REVIEW

**Comment by Sayonarko** [comment]
[BugFix live page buttons: Change emoji  for buttons rec, pause, stop into icons](https://trello.com/c/rGqLaqz3/240-bugfix-live-page-buttons-change-emoji-for-buttons-rec-pause-stop-into-icons)

**Comment by Sayonarko** [comment]
[Add follow user functionality](https://trello.com/c/Agbez49X/228-add-follow-user-functionality)

File: backend/src/services/user-follower/user-follower.service.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  constructor({ userFollowerRepository }: Constructor) {
+    this.#userFollowerRepository = userFollowerRepository;
+  }
+
+  public getCountByUserId(userId: number): Promise<number> {
+    return this.#userFollowerRepository.getCountByUserId(userId);
+  }
+
+  public isFollowed(payload: UserFollowerPayload): Promise<TUserFollower> {
```
Comment (what1s1ove) [inline backend/src/services/user-follower/user-follower.service.ts]:
```suggestion
  public checkIsFollowed(payload: UserFollowerPayload): Promise<TUserFollower> {
```
functions with this naming should return boolean

File: frontend/src/store/user-profile/actions.ts [hunk 2]
```diff
@@ -22,4 +22,37 @@ const loadUser = createAsyncThunk<User, number, AsyncThunkConfig>(
...
 
-export { loadPodcasts, loadUser };
+const getFollowersCount = createAsyncThunk<number, number, AsyncThunkConfig>(
+  ActionType.GET_FOLLOWERS_COUNT,
+  async (userId, { extra }) => {
+    const { userFollowerApi } = extra;
+    return await userFollowerApi.getCountByUserId(userId);
+  });
+
+const isFollowedUser = createAsyncThunk<boolean, UserFollowerPayload, AsyncThunkConfig>(
```
Comment (what1s1ove) [inline frontend/src/store/user-profile/actions.ts]:
```suggestion
const checkIsFollowedUser = createAsyncThunk<boolean, UserFollowerPayload, AsyncThunkConfig>(
```

File: frontend/src/store/user-profile/actions.ts [hunk 3]
```diff
@@ -22,4 +22,37 @@ const loadUser = createAsyncThunk<User, number, AsyncThunkConfig>(
...
+  },
+);
+
+const unfollowUser = createAsyncThunk<void, UserFollowerPayload, AsyncThunkConfig>(
+  ActionType.UNFOLLOW_USER,
+  async (payload, { extra }) => {
+    const { userFollowerApi } = extra;
+    await userFollowerApi.delete(payload);
+  },
+);
```
Comment (what1s1ove) [inline frontend/src/store/user-profile/actions.ts]: rewrite it into one action `toggleUserFollow`

File: frontend/src/services/user-follower-api/user-follower-api.service.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  public getCountByUserId(userId: number): Promise<number> {
+    return this.#http.load(
+      `${this.#apiPrefix}${ApiPath.USER_FOLLOWERS}/${userId}`,
+      {
+        method: HttpMethod.GET,
+      },
+    );
+  }
+
+  public isFollowed(payload: UserFollowerPayload): Promise<UserFollower> {
```
Comment (what1s1ove) [inline frontend/src/services/user-follower-api/user-follower-api.service.ts]:
```suggestion
  public checkIsFollowed(payload: UserFollowerPayload): Promise<UserFollower> {
```
functions with this name should return boolean

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/api/user-followers/user-followers.api.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  userFollowerRouter.get(
+    UserFollowersApiPath.$USER_ID_$FOLLOWER_ID,
+    checkAuthMiddleware(HttpMethod.GET),
+    handleAsyncApi(async (req, res) => {
+      const userFollower = await userFollowerService.isFollowed({
+        userId: Number(req.params.userId),
+        followerId: Number(req.params.followerId),
+      });
+
+      return res.json(userFollower || null).status(HttpCode.OK);
```
Comment (what1s1ove) [inline backend/src/api/user-followers/user-followers.api.ts]:
```suggestion
      return res.json(userFollower ?? null).status(HttpCode.OK);
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**